0011-command-execution-protocol/README.md

```diff
@@ -0,0 +1,170 @@
+
+This document describes how commands are delivered to endpoints and how the server is notified of command execution results. This document does not cover who and how initiates command, and how results are interpreted.
+
+# Language
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+- **Command** is an action requested to be executed on a specific endpoint that may produce a result or fail.
+- **Command Invocation** is an act of a command entering the system and being scheduled for execution.
+- **Command Execution** is an act of transfering a command to endpoint for actual execution and waiting for execution results.
+
+# Requirements and constraints
+- Command execution is asynchronous in nature and may take time. Furthermore, the endpoint may reboot or client may reconnect as a part of the command execution sequence.
+- An endpoint may support multiple command types, thus it must tell them apart.
+- Command may success or fail. The caller must know outcomes of the command or the reason of a failure.
+- Commands are often not idempotent. That is, executing a single command twice may result in a different (possibly undesired) outcome.
+- An endpoint may be offline at the time of command invocation. The caller must be informed that the command execution did not succeed.
+
+# Use cases
+## UC1: Remote control
+An endpoint operates a device. Command Execution Protocol may be used to transfer commands to the endpoint and instruct it to change device behavior.
+
+## UC2: Expose local resources
+An endpoint represents a sensor that does not measure the value periodically. Command Execution Protocol may be used to issue a measurement command to the endpoint and transfer measurement results back to the caller.
+
+## UC3: OTA update
+> Note: while OTA updates require more support and probably a separate protocol, a basic version can be implemented using command execution.
+
+An endpoint may receive a command to update its firmware. When update is complete, the endpoint sends an update result to the server.
+
+# Design
+The Command Execution Protocol extension consists of two resource sets with the corresponding resource paths:
+- `/<endpoint_token>/command/<command_type>` - command resource
+- `/<endpoint_token>/result/<command_type>` - result resource
+
+## Command resource
+Command resource is used to distribute commands to the endpoints.
+
+The command resource is a request/response resource with an observe extension which resides at the following resource path:
+```
+/<endpoint_token>/command/<command_type>
+```
+where `<command_type>` is a type of the command the endpoint wants to serve.
+
+Command type MUST be a non-empty alpha-numeric string identifying the command type to the endpoint.
+
+### Request
```

Comment:
I'd rename to "Command request" so that there'd be no anchor ambiguity on this page. It makes the content a little clearer, too.